### PR TITLE
fix RoF bug, cross references were deleted when moving to flash in rd…

### PR DESCRIPTION
**Pull request by OfirMos**
…b_save, the fix: introduced new flag which states if the series is going to flash or not

### CODE REVIEW

**Comment by OfirMos** [comment]
Please pay attention to this:
`__attribute__((weak)) int (*RedisModule_SetDataTypeExtensions)(
    RedisModuleCtx *ctx,
    RedisModuleType *mt,
    RedisModuleTypeExtMethods *typemethods) REDISMODULE_ATTR = NULL;`

**Review by danni-m — APPROVED** [review]

**Comment by MeirShpilraien** [comment]
> Please pay attention to this: `__attribute__((weak)) int (*RedisModule_SetDataTypeExtensions)( RedisModuleCtx *ctx, RedisModuleType *mt, RedisModuleTypeExtMethods *typemethods) REDISMODULE_ATTR = NULL;`

Can you explain why we need to pay attention to it? Also state it on the commit message..

File: src/module.c [hunk 1]
```diff
@@ -1158,6 +1158,26 @@ void swapDbEventCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub,
...
+    series->in_ram = true;
+}
+
+int keyRemovedFromDbDict(RedisModuleCtx *ctx,
+                         RedisModuleString *key,
+                         void *value,
+                         int swap_key_metadata,
+                         int writing_to_swap) {
+    Series *series = (Series *)value;
+    if (!!writing_to_swap) {
```
Comment (MeirShpilraien) [inline src/module.c:1175]: Can you add a comment what this variable means and why we need to change the flag only if this variable is true?

File: src/module.c [hunk 2]
```diff
@@ -1158,6 +1158,26 @@ void swapDbEventCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub,
...
+        series->in_ram = false;
+    }
+    return 0;
```
Comment (MeirShpilraien) [inline src/module.c:1178]: Whats the return value means for RoF?

File: src/module.c [hunk 3]
```diff
@@ -1307,6 +1327,11 @@ void Initialize_RdbNotifications(RedisModuleCtx *ctx) {
     }
 }
 
+__attribute__((weak)) int (*RedisModule_SetDataTypeExtensions)(
```
Comment (MeirShpilraien) [inline src/module.c:1330]: Explain why you need this

File: src/module.c [hunk 4]
```diff
@@ -1382,6 +1407,17 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
     SeriesType = RedisModule_CreateDataType(ctx, "TSDB-TYPE", TS_LATEST_ENCVER, &tm);
     if (SeriesType == NULL)
         return REDISMODULE_ERR;
+
+    RedisModuleTypeExtMethods etm = { .version = REDISMODULE_TYPE_EXT_METHOD_VERSION,
+                                      .key_added_to_db_dict = keyAddedToDbDict,
+                                      .removing_key_from_db_dict = keyRemovedFromDbDict,
+                                      .get_key_metadata_for_rdb = NULL };
+    if (RedisModule_SetDataTypeExtensions != NULL) {
```
Comment (MeirShpilraien) [inline src/module.c:1415]: I believe we have a macro to check if API is available.

**Review by MeirShpilraien — COMMENTED** [review]
I believe we can improve the commit message and explain what was the issue and what is the fix.

**Pull request merged**